app/controllers/restaurants_controller.rb

```diff
@@ -1,5 +1,50 @@
 class RestaurantsController < ApplicationController
+  before_action :require_restaurant, only: %i[update edit destroy]
+  
+  def index
+    @restaurants = current_user.restaurants.all
+  end
 
-  def index; end
+  def new
+    @restaurant = Restaurant.new
+  end
+
+  def create
+    @restaurant = current_user.restaurants.new(restaurant_params)
+    if @restaurant.save
+      redirect_to restaurants_path
+    else
+      flash[:register_errors] = @restaurant.errors.full_messages
+      redirect_to new_restaurant_path
+    end
+  end
+    
+  def edit
+    if @restaurant.nil?
```

Comment:
You can handle this condition in filter so it will help's for all three actions !